src/config.rs

```diff
@@ -59,19 +79,37 @@ impl From<toml::DecodeError> for Error {
     }
 }
 
+impl Decodable for GpioConfig {
+    fn decode<D: Decoder>(d: &mut D) -> Result<Self, D::Error> {
+        // Get items under the [config] header if present
+        let symlink_root: Option<String> = d.read_struct_field("config", 0, |cfg| {
+                                                cfg.read_struct_field("symlink_root",
+                                                                      0,
+                                                                      Decodable::decode)
+                                            })
+                                            .ok();
+
+        Ok(GpioConfig {
+            pins: try!(d.read_struct_field("pins", 0, Decodable::decode)),
+            symlink_root: symlink_root,
+        })
+    }
+}
+
 impl Decodable for PinConfig {
     fn decode<D: Decoder>(d: &mut D) -> Result<PinConfig, D::Error> {
         Ok(PinConfig {
             num: try!(d.read_struct_field("num", 0, Decodable::decode)),
             direction: d.read_struct_field("direction", 0, |dir_d| {
-                match &try!(dir_d.read_str())[..] {
-                    "in" => Ok(sysfs_gpio::Direction::In),
-                    "out" => Ok(sysfs_gpio::Direction::Out),
-                    "high" => Ok(sysfs_gpio::Direction::High),
-                    "low" => Ok(sysfs_gpio::Direction::Low),
-                    _ => Err(dir_d.error("Expected one of: {in, out, high, low}")),
-                }
-            }).unwrap_or(sysfs_gpio::Direction::In), // default: In
+                            match &try!(dir_d.read_str())[..] {
```

Comment:
What is the `[..]` operator doing here? I'm not following why it's needed, since you should already have a `&str`...